programs/client/Client.cpp

```diff
@@ -1120,7 +1121,7 @@ void Client::processOptions(const OptionsDescription & options_description,
     if (options.count("config"))
         config().setString("config-file", options["config"].as<std::string>());
     if (options.count("host") && !options["host"].defaulted())
-        config().setString("host", options["host"].as<std::string>());
+        hosts = options["host"].as<std::vector<std::string>>();
```

Comment:
Seems like you need to set config here, but not in the beginning of `Client::connect`

Reply:
Setting `config()` is a crutch. My purpose is to save hosts anywhere, after that collegue can use this vector to make connection. But you are right, it would be better, if this crutch will be there, not in beginnin of `connect`.

Reply:
https://github.com/ManagedDatabases/ClickHouse/pull/4#discussion_r743621945